app/scripts/modules/amazon/serverGroup/details/serverGroupDetails.aws.controller.js

```diff
@@ -418,5 +425,46 @@ module.exports = angular.module('spinnaker.serverGroup.details.aws.controller',
         serverGroup.accountDetails = details;
       });
     };
+
+    this.lookForPipelineCommitDiffs = (executionId) => {
+      if (executionId) {
+        executionService.getExecution(executionId).then(executionDetails => {
+
+          this.jarDiffs = {
+            added: [],
+            downgraded: [],
+            duplicates: [],
+            removed: [],
+            unchanged: [],
+            unknown: [],
+            upgraded: []
+          };
+          this.commits = [];
+
+          const deployStages = executionDetails.stages.filter(stage => stage.type === 'deploy');
```

Comment:
Using `stageId` works and GREATLY reduces the amount of code needed to process the execution details.  Good tip you guys.  Thanks.